src/vm/fastserializer.cpp

```diff
@@ -198,7 +198,7 @@ void FastSerializer::WriteBuffer(BYTE *pBuffer, unsigned int length)
         // This will cause us to stop writing to the file.
         // The file will still remain open until shutdown so that we don't
         // have to take a lock at this level when we touch the file stream.
-        m_writeErrorEncountered = (length != outCount);
```

Comment:
On Windows - this is probably okay. Note that the pipe is opened with the PIPE_WAIT option and therefore the Write call will block until the client frees up the buffer.

On Unix - this is not acceptable - the underlying system call could be interrupted and therefore write less than required - and that is NOT an error condition.

Reply:
From the [Linux Programmer's Manual](http://man7.org/linux/man-pages/man2/send.2.html):
```log
When the message does not fit into the send buffer of the socket,
send() normally blocks, unless the socket has been placed in
nonblocking I/O mode.  In nonblocking mode it would fail with the
error EAGAIN or EWOULDBLOCK in this case.
```

Reply:
>Note that a successful write() may transfer fewer than count bytes.
       Such partial writes can occur for various reasons; for example,
       because there was insufficient space on the disk device to write all
       of the requested bytes, or because a blocked write() to a socket,
       pipe, or similar was interrupted by a signal handler after it had
       transferred some, but before it had transferred all of the requested
       bytes.  In the event of a partial write, the caller can make another
       write() call to transfer the remaining bytes.  The subsequent call
       will either transfer further bytes or may result in an error (e.g.,
       if the disk is now full).
> - write(2) man page

So we should check if we wrote everything and attempt again if we wrote less than we wanted.